### DISCUSSION TITLE
Are there optimizations for linking only the neccesary libs intead of every lib in `vcpkg`?

### DISCUSSION THREAD

**Discussion by playgithub** [microsoft/vcpkg#24137] [Ideas]
Are there optimizations for linking only the neccesary libs intead of every lib in `vcpkg`?

**Comment by autoantwort** [comment]
You can manually link only these libs that are used by your project in msbuild. But it is impossible for vcpkg to detect which libs are actually needed by your project.

**Comment by playgithub** [comment]
> You can manually link only these libs that are used by your project in msbuild.

It'll be boring when there are massive transitive dependencies.

> But it is impossible for vcpkg to detect which libs are actually needed by your project.

Is it better if top depencies are set manually in settings as below and transitive dependencies are set automatically.

![image](https://user-images.githubusercontent.com/3665590/163200932-728f8538-ac59-4765-9465-1b97baa0e144.png)
